check-ssl-host plugin

mivok opened the pull request:
Connects to a HTTPS (or other SSL) server and performs several checks on the certificate:

- Is the hostname valid for the host we're requesting
- If any certificate chain is presented, is it valid (i.e. is each certificate signed by the next)
- Is the certificate about to expire

There are a couple of other plugins that work with SSL certificate, but other plugins focus mostly on expiry, and don't support SNI (except for the net/http based ones in ruby 2.0 and above).

mattyjones commented:
@mivok Sorry for the long wait on this.  Could you please update the header to the standard one found [here](https://github.com/sensu/sensu-community-plugins/blob/master/CONTRIBUTING.md) and then I will merge this.

Thanks

mivok commented:
@mattyjones No worries, thanks for taking a look. I've updated the header (and added a missing require that I don't recall being needed when I was testing previously). Let me know if you'd prefer all of this rebasing into a single commit, or if things are good as is.

Thanks.

mattyjones commented:
@mivok Things are fine as is.  I prefer rebasing to avoid merge commits but this is fine.  I will look at merging this tonight.

Thanks!
